packages/hardhat-chai-matchers/src/internal/errors.ts

```diff
@@ -23,3 +23,11 @@ export class HardhatChaiMatchersAssertionError extends HardhatChaiMatchersError
     super(`Assertion error: ${message}`);
   }
 }
+
+export class HardhatChaiMatchersNonChainableMatcherError extends HardhatChaiMatchersError {
+  constructor(matcherName: string) {
+    super(
+      `${matcherName} is not chainable. Please refer to the [README](https://github.com/NomicFoundation/hardhat/blob/main/packages/hardhat-chai-matchers#chaining-async-matchers) for more information.`
```

Comment:
We can't use markdown here. The way we normally do this is to add a shortlink like [this](https://github.com/NomicFoundation/hardhat/blob/f324b3a335587d24da3c1901ed2b5728d6e4cebb/docs/redirects.config.js#L64-L68) and then use that in the error message.